mlflow/transformers/__init__.py

```diff
@@ -1183,8 +1175,22 @@ def _load_model(path: str, flavor_config, return_type: str, device=None, **kwarg
         conf["device_map"] = device_map_strategy
         accelerate_model_conf["device_map"] = device_map_strategy
         # Cannot use device with device_map
+        if device is not None:
+            _logger.warning(
+                "MLFLOW_HUGGINGFACE_USE_DEVICE_MAP is set to True, device "
+                f"is incompatible with device_map, ignoring device param `{device}` passed."
+            )
         device = None
 
+    if device is None:
+        if MLFLOW_DEFAULT_PREDICTION_DEVICE.get():
+            try:
+                device = int(MLFLOW_DEFAULT_PREDICTION_DEVICE.get())
+            except ValueError:
+                device = _TRANSFORMERS_DEFAULT_CPU_DEVICE_ID
+        elif is_gpu_available():
+            device = _TRANSFORMERS_DEFAULT_GPU_DEVICE_ID
```

Comment:
Doesn't this change the behavior when `MLFLOW_HUGGINGFACE_USE_DEVICE_MAP` is set to `True`?Previously, `device` is set to `None`, however, now it is overwritten here and one of these default values will be passed as `device` key of the loading config. Is it intended?

Reply:
Oh right! Thanks for capturing this! Let me add a test